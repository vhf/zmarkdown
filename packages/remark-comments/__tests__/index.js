import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'

const render = text => unified()
  .use(reParse)
  .use(plugin)
  .use(remark2rehype)
  .use(stringify)
  .processSync(text)

test('comments', () => {
  const { contents } = render(dedent`
    Test comments
    =============

    Foo<--COMMENTS I will be gone COMMENTS-->bar

    \`\`\`
    Foo<--COMMENTS I will not get remove because I am in a code block COMMENTS-->bar
    \`\`\`

    <--COMMENTS Unfinished block won't get parsed either
  `)
  expect(contents).toMatchSnapshot()
})
