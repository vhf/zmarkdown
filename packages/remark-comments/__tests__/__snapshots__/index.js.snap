// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments 1`] = `
"<h1>Test comments</h1>
<p>Foobar</p>
<pre><code>Foo&#x3C;--COMMENTS I will not get remove because I am in a code block COMMENTS-->bar
</code></pre>
<p>&#x3C;--COMMENTS Unfinished block won't get parsed either</p>"
`;
